test/FluentMigrator.Tests/Integration/IntegrationTestBase.cs

```diff
@@ -90,92 +64,19 @@ public void TearDownDataDirectory()
-        {
-            foreach (var (processorType, getOptionsFunc) in _processors)
-            {
-                var opt = getOptionsFunc();
-                if (!opt.IsEnabled)
-                    continue;
-
-                if (!isMatch(processorType))
-                    continue;
-
-                return true;
-            }
-
-            return false;
-        }
-
-        protected void ExecuteWithSupportedProcessors(
-            Action<IServiceCollection> initAction,
-            Action<IServiceProvider, ProcessorBase> testAction,
-            bool tryRollback = true,
-            params Type[] exceptProcessors)
-        {
-            ExecuteWithSupportedProcessors(
-                initAction,
-                testAction,
-                tryRollback,
-                procType => !exceptProcessors.Any(p => p.IsAssignableFrom(procType)));
-        }
-
-        private void ExecuteWithSupportedProcessors(
-            Action<IServiceCollection> initAction,
-            Action<IServiceProvider, ProcessorBase> testAction,
-            bool tryRollback,
-            Predicate<Type> isMatch)
-        {
-            if (!IsAnyServerEnabled())
-            {
-                Assert.Fail(
-$"No database processors are configured to run your migration tests.  This message is provided to avoid false positives.  To avoid this message enable one or more test runners in the {nameof(IntegrationTestOptions)} class.");
-            }
-
-            var executed = false;
-            foreach (var (processorType, getOptionsFunc) in _processors)
-            {
-                var opt = getOptionsFunc();
-                if (!opt.IsEnabled)
-                    continue;
-
-                if (!isMatch(processorType))
-                    continue;
-
-                executed = true;
-                ExecuteWithProcessor(processorType, initAction, testAction, tryRollback, opt);
-            }
-
-            if (!executed)
-            {
-                Assert.Ignore("No processor found for the given action.");
-            }
-        }
-
-        protected void ExecuteWithProcessor<TProcessor>(
-            Action<IServiceCollection> initAction,
-            Action<IServiceProvider, TProcessor> testAction,
-            bool tryRollback,
-            IntegrationTestOptions.DatabaseServerOptions serverOptions)
-            where TProcessor : ProcessorBase
-        {
-            ExecuteWithProcessor(typeof(TProcessor), initAction, (sp, proc) => testAction(sp, (TProcessor)proc), tryRollback, serverOptions);
-        }
-
-        private void ExecuteWithProcessor(
+        protected void ExecuteWithProcessor(
             Type processorType,
             Action<IServiceCollection> initAction,
-            Action<IServiceProvider, ProcessorBase> testAction,
-            bool tryRollback,
-            IntegrationTestOptions.DatabaseServerOptions serverOptions)
+            Action<IServiceProvider, GenericProcessorBase> testAction,
```

Comment:
Just wondering... Why GenericProcessorBase instead of ProcessorBase?

Reply:
Now I wonder if that's the best way to do it: in a few tests (SQL server tests, in fact), I use CleanupTestDatabase (previously named CleanupTestSqlServerDatabase, and it took a TProcessor of type SqlServerProcessor), and but we use the GenericProcessorBase.Connection in it, which is only declared in GenericProcessorBase.

I'm thinking I could do this instead and let the testAction type as it was :

    if (origProcessor is GenericProcessorBase gpb)
    {
        gpb.Connection.Close();
    }

Reply:
There are two scenarios: 

a Processor that implements a Connection and a Processor that does not implement a Connection. cf ConnectionlessProcessor https://github.com/fluentmigrator/fluentmigrator/blob/ed292797262ba098fbc09ab4ea2944062fc723fe/src/FluentMigrator.Runner.Core/Processors/ConnectionlessProcessor.cs#L37

Reply:
Ideally, Processor would not "know" about its connection directly, as it creates too much coupling between the processor meta-level checks that run queries, and the connection that executes those commands.

Reply:
Ok so is that ok if I revert the change and do this ? 
![image](https://github.com/user-attachments/assets/b1f5d26f-96c7-45a2-be6d-0dd67c8fae07)

Reply:
By the way I was not aware of the connectionless processor, that could be useful (but it does not inherit ProcessorBase though), I could change ExecuteWithProcessor with `Action<IServiceProvider, IMigrationProcessor> testAction`maybe ?